client.go

```diff
@@ -219,6 +281,49 @@ func (c *Client) sendError(err error) {
 	}
 }
 
+// CapRequest allows you to request IRCv3 capabilities from the server during
+// the handshake. The behavior is undefined if this is called before the
+// handshake completes so it is recommended that this be called before Run. If
+// the CAP is marked as required, the client will exit if that CAP could not be
+// negotiated during the handshake.
+func (c *Client) CapRequest(capName string, required bool) {
+	cap := c.caps[capName]
+	cap.Requested = true
+	cap.Required = cap.Required || required
```

Comment:
should you log if  `cap.Required` is false and you're explicitly marking it as true?

Reply:
Probably not. If one plugin requests it as optional and one plugin requests it as required, it should be marked as required. Plus the logging in go-irc is somewhat limited on purpose.